Random_numbers/include/CGAL/Random.h

```diff
@@ -95,6 +96,15 @@ class Random {
     return generator();
   }
 
+  // Overload to avoid a warning with VC++
+  inline
+  std::size_t
+  uniform_smallint(std::size_t lower, std::size_t upper)
+  {
+    return uniform_smallint(static_cast<boost::rand48::result_type>(lower),
+                            static_cast<boost::rand48::result_type>(upper));
+  }
+
```

Comment:
That patch causes an infinite recursion if `std::size_t` and `boost::rand48::result_type` are the same type! Mesh_3 got hit on all 32 bits platforms:

https://cgal.geometryfactory.com/CGAL/testsuite/results-4.12-Ic-180.shtml#Mesh_3

Why not casting `lower` and `upper` directly in the template version of `uniform_smallint`?